pkg/eventcounter/api/api_test.go

```diff
@@ -919,6 +921,200 @@ func TestGenInterval(t *testing.T) {
 	}
 }
 
+func TestGetOneMonthTimeStamps(t *testing.T) {
+	t.Parallel()
+
+	endAt := time.Now()
+	startAt, err := genInterval(jpLocation, endAt, 30)
+	assert.NoError(t, err)
+
+	patterns := []struct {
+		desc             string
+		startAt          time.Time
+		expectedElements []int64
+		expectedLen      int
+	}{
+		{
+			desc:    "success",
+			startAt: startAt,
+			expectedElements: []int64{
+				getDate(startAt),
+				getDate(startAt.AddDate(0, 0, 1)),
+				getDate(endAt),
+			},
+			expectedLen: 31,
+		},
+	}
+	for _, p := range patterns {
+		t.Run(p.desc, func(t *testing.T) {
+			actual := getOneMonthTimeStamps(p.startAt)
+			for _, e := range p.expectedElements {
+				assert.Contains(t, actual, int64(e))
+			}
+			assert.Len(t, actual, p.expectedLen)
+		})
+	}
+}
+
+func TestGetVariationIDs(t *testing.T) {
+	t.Parallel()
+
+	vID1 := newUUID(t)
+	vID2 := newUUID(t)
+
+	patterns := []struct {
+		desc       string
+		variations []*featureproto.Variation
+		expected   []string
+	}{
+		{
+			desc: "success",
+			variations: []*featureproto.Variation{
+				{
+					Id:    vID1,
+					Value: "true",
+				},
+				{
+					Id:    vID2,
+					Value: "false",
+				},
+			},
+			expected: []string{
+				vID1, vID2, defaultVariationID,
+			},
+		},
+	}
+	for _, p := range patterns {
+		t.Run(p.desc, func(t *testing.T) {
+			actual := getVariationIDs(p.variations)
+			assert.Equal(t, actual, p.expected)
+		})
+	}
+}
+
+func TestGetEvaluationTimeseriesCountV2(t *testing.T) {
+	t.Parallel()
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+
+	patterns := []struct {
+		desc        string
+		setup       func(*eventCounterService)
+		input       *ecproto.GetEvaluationTimeseriesCountRequest
+		expected    *ecproto.GetEvaluationTimeseriesCountResponse
+		expectedErr error
+	}{
+		{
+			desc: "error: ErrFeatureIDRequired",
+			input: &ecproto.GetEvaluationTimeseriesCountRequest{
+				EnvironmentNamespace: "ns0",
+			},
+			expectedErr: localizedError(statusFeatureIDRequired, locale.JaJP),
+		},
+		{
+			desc: "success",
+			setup: func(s *eventCounterService) {
+				s.featureClient.(*featureclientmock.MockClient).EXPECT().GetFeature(gomock.Any(), gomock.Any()).Return(
```

Comment:
Let's set the arguments for `GetFeature` to ensure that it is passing correctly instead of setting `gomock.Any()`.
Same for `GetEventCounts` and `GetUserCounts`.

E.g.
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/eventpersister/persister/persister_test.go#L382-L389

Reply:
>Same for GetEventCounts and GetUserCounts.

We can't do that because the parameters always change in for loop.

Reply:
Since you are using slices, it can be done by looping the mock interface.
The mock interface can be called as much as needed. You just need to ensure that you call it in order.

- Create the test keys for each variation for `EventCount`.
- Create a slice that contains the test values to be returned for which key
- Loop the mock `GetEventCounts` interface and set the keys and the values to be returned

Doing this will ensure that the keys passing to the cacher, the returned values for which key, and the number of calls are correct.

E.g.
```go
keys := []string{...}
values := []string{...}
for i := 0; i < len(keys); i++ {
  s.evaluationCountCacher.(*eccachemock.MockEventCounterCache).EXPECT().GetEventCounts(keys[i]).Return([]float64{values[i]}, nil)
}
```

The same thing can be done for `UserCount`.

Reply:
Thank you. I was able to implement it 👍